### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by cmsirbu**
## What's changed

- Moved existing documentation into a new structure to prepare for the mkdocs rendered docsite.
- Edited and improved the visuals in various places using new syntax from the [Material for Mkdocs theme](https://squidfunk.github.io/mkdocs-material/reference/).
- Added dependencies to poetry environment for development.
- Added configuration for building and hosting the docsite on RTD.
- Replaced the project README with a standardized one. Note: the README will have certain links/images broken until the PR is merged into `develop`.
- Added `invoke docs` task to aid serving docs locally in the dev environment.
- Added test to check that RTD used `docs/requirements.txt` are in sync with dependencies in `pyproject.toml`.

## Building docs

You can build the docs locally in the poetry environment:

```
poetry install
poetry run mkdocs serve
```

See below notes on development environment (more specifically the docker based one).

## TODO

Some sections throughout the files listed below require input (or removal if not applicable to this plugin). All `Developer Note` comments should be resolved and removed.
- [x] user/app_overview.md
- [x] user/app_use_cases.md - you might want to add more screenshots
- [x] user/app_getting_started.md
- [x] user/faq.md
- [x] user/external_interactions.md
- [x] admin/install.md - review
- [x] admin/upgrade.md
- [x] admin/uninstall.md
- [x] admin/compatiblity_matrix.md
- [x] admin/release_notes - I think the CHANGELOG content should be linking to the docs section release notes, but will leave this decision up to you.
- [x] dev/extending.md - review
- [x] dev/contributing.md
- [x] dev/dev_environment.md - boilerplate guide based on cookie template (old GETTING_STARTED guide). A sync to the cookie dev environment would be beneficial to ensure instructions are accurate.
- [x] dev/code_reference - check examples provided and add any module/package auto-generated docs that are relevant
- [x] README.md - review

### CODE REVIEW

File: docs/user/app_getting_started.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+This document provides a step-by-step tutorial on how to get the App going and how to use it.
+
+## Install the App
+
+To install the App, please follow the instructions detailed in the [Installation Guide](../admin/install.md).
+
+## First steps with the App
+
+!!! warning "Developer Note - Remove Me!"
```
Comment (chadell) [inline docs/user/app_getting_started.md:11]: @scetron "Remove Me", also below

File: docs/user/app_use_cases.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+### Define source emails per Provider
+
+In the Nautobot UI, under **Circuits -> Providers**, for each Provider that we would like to track via the Circuit Maintenance plugin, we **must** configure at least one email source address (or a comma-separated list of addresses) in the **`Custom Fields -> Emails for Circuit Maintenance plugin** field.
+
+These are the source email addresses that the plugin will detect and will use to classify each notification for each specific provider.
+
+Also, by default, the Provider **slug** is used to match the provider parser from the `circuit-maintenance-parser` library, but if a custom mapping is desired (i.e. CenturyLink to Lumen), you can define this custom mapping in the **`Custom Fields -> Provider Parser for Circuit Maintenance plugin** field.
+
+### Configure Notification Sources in `nautobot_config.py`
```
Comment (chadell) [inline docs/user/app_use_cases.md:15]: this to me looks more like "getting started"

File: docs/user/external_interactions.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# External Interactions
```
Comment (chadell) [inline docs/user/external_interactions.md:1]: maybe the notification source should come here? those are external integrations

**Review by chadell — CHANGES_REQUESTED** [review]

**Comment by cmsirbu** [comment]
You can see the docs as-built from this branch live here: https://docs.nautobot.com/projects/circuit-maintenance/en/latest/

There are some leftover developer notes in the README.md - you might want to run something like `rgrep -i "developer note"` on the repo to catch'em all.

Note: the logo in the README will be broken until this is merged.

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**